### DISCUSSION TITLE
Ignore line breaks for MD or ReST files?

### DISCUSSION THREAD

**Discussion by nodet** [dandavison/delta#1723] [Q&A]
Is there a way to configure Delta such that it will ignore line breaks where they don't matter, such as in most paragraphs of Markdown or reStructuredText files.

![image](https://github.com/dandavison/delta/assets/608561/5dc02a7b-edf5-4ef7-895c-6943e302af75)

We want such paragraphs to have line breaks in the source files. But when these paragraphs get reflowed, this can lead to large diffs that mostly consist of words moving from one line to the next or previous.  This makes it harder than necessary to review the changes.
Delta already helps, when compared with a raw `git diff`. But if it would entirely ignore the line breaks, it would be even better.  Did I miss some configuration?  Was this ever considered?

**Comment by dandavison** [comment]
Hi @nodet, the thing is that delta doesn't compute the diff; delta is just a  renderer for a diff computed by git or `diff`. What you want is achievable I think -- by transforming the text (e. g. with sed, or ripgrep) in a pipeline before git's diff engine sees it. I'm afk right now and I forget what it's called, but IIRC git has such a facility -- I used it once a long time ago to diff MSWord docs by turning them into XML first...

**Reply by nodet** [reply]
It seems you're referring to the [diff attribute](https://git-scm.com/docs/gitattributes#_generating_diff_text). That indeed looks interesting, thanks!
